File: nodejs/csv2json/processor.js
import fs from "fs";
import { parse } from "csv-parse";

export async function processFile(inFile, outFile, processItem) {
  const inStream = fs.createReadStream(inFile).pipe(
    parse({
      columns: true,
      relax_quotes: true,
    })
  );

  const outStream = fs.createWriteStream(outFile);

  for await (const inItem of inStream) {
    const outItem = processItem(inItem);
    outStream.write(JSON.stringify(outItem) + "\n");
  }
}

File: python/csv2json/processor.py
import csv
import json


def process_file(in_file_path, out_file_path, process_item):
    with open(in_file_path, 'r') as in_file:
        reader = csv.DictReader(in_file)
        with open(out_file_path, 'w', newline='') as out_file:
            for in_data in reader:
                out_data = process_item(in_data)
                json.dump(out_data, out_file)
                out_file.write('\n')
